File: clients/javascript/src/models/authTokenCreateOut.ts
// this file is @generated

export interface AuthTokenCreateOut {
    id: string;
    createdAt: Date;
    updatedAt: Date;
    token: string;
}

export const AuthTokenCreateOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AuthTokenCreateOut {
        return {
            id: object['id'],
            createdAt: new Date(object['created_at']),
            updatedAt: new Date(object['updated_at']),
            token: object['token'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AuthTokenCreateOut): any {
        return {
            'id': self.id,
            'created_at': self.createdAt,
            'updated_at': self.updatedAt,
            'token': self.token,
        };
    }
}

File: clients/rust/src/models/auth_token_create_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AuthTokenCreateOut {
    pub id: String,

    pub created_at: jiff::Timestamp,

    pub updated_at: jiff::Timestamp,

    pub token: String,
}

impl AuthTokenCreateOut {
    pub fn new(
        id: String,
        created_at: jiff::Timestamp,
        updated_at: jiff::Timestamp,
        token: String,
    ) -> Self {
        Self {
            id,
            created_at,
            updated_at,
            token,
        }
    }
}
